docs: Permission segregation for Azure AD Pod & Workload Identity. (#790)
kedacore/keda-docs: content/docs/2.8/concepts/authentication.md

spec:
  podIdentity:
      provider: none | azure | azure-workload | aws-eks | aws-kiam  # Optional. Default: none

>       identityID: <identity-id>                         # Optional. Only used by azure & azure-workload providers.

Reviewer: Nit, we typically use `Id` instead of `ID`
```suggestion
      identityId: <identity-id>                         # Optional. Only used by azure & azure-workload providers.
```

Reviewer: Same for other places, obviously.

```yaml
podIdentity:
  provider: azure           # Optional. Default: none

>   identityID: <identity-id> # Optional. Default: Identity linked with the label set when installing KEDA. Refer to Pod Identity docs, and the docs below.

Reviewer: This is already the pod identity section so would leave this part out
```suggestion
  identityID: <identity-id> # Optional. Default: Identity linked with the label set when installing KEDA.
```

```

Azure AD Pod Identity will give access to containers with a defined label for `aadpodidbinding`.  You can set this label on the KEDA operator deployment.  This can be done for you during deployment with Helm with `--set podIdentity.activeDirectory.identity={your-label-name}`.

> You can override the identity set while installing KEDA by specifying an `identityID` in the `podIdentity` field. This can be used to prevent an Azure Managed Identity from having permissions to multiple Azure resources, due to security
> concerns.

Reviewer: How about this?
```suggestion
You can override the identity that was assigned to KEDA while installing it by specifying an `identityID` in the `podIdentity` field. This allows end-users to use different identities that have access to various resources which is more secure compared to only using one identity that has access to multiple Azure resources.
```

```yaml
podIdentity:
  provider: azure           # Optional. Default: none
  identityId: <identity-id> # Optional. Default: Identity linked with the label set when installing KEDA.
```

Azure AD Pod Identity will give access to containers with a defined label for `aadpodidbinding`.  You can set this label on the KEDA operator deployment.  This can be done for you during deployment with Helm with `--set podIdentity.activeDirectory.identity={your-label-name}`.

> You can override the identity that was assigned to KEDA during installation, by specifying an `identityId` parameter under
> the `podIdentity` field. This allows end-users to use different identities to access various resources which is more secure than using a single identity that has access to multiple resources.

Reviewer: No need to break lines.
 ```suggestion
You can override the identity that was assigned to KEDA during installation, by specifying an `identityId` parameter under the `podIdentity` field. This allows end-users to use different identities to access various resources which is more secure than using a single identity that has access to multiple resources.
```

2. `--set podIdentity.azureWorkload.clientId={azure-ad-client-id}`
3. `--set podIdentity.azureWorkload.tenantId={azure-ad-tenant-id}`

> You can override the clientId set while installing KEDA by specifying an `identityID` in the `podIdentity` field. This can be used to prevent an Azure AD Application from having permissions to multiple Azure resources, due to security

Reviewer: See comment above

2. `--set podIdentity.azureWorkload.clientId={azure-ad-client-id}`
3. `--set podIdentity.azureWorkload.tenantId={azure-ad-tenant-id}`

> You can override the identity that was assigned to KEDA during installation, by specifying an `identityId` parameter under
> the `podIdentity` field. This allows end-users to use different identities to access various resources which is more secure than using a single identity that has access to multiple resources.

Reviewer: No need to break lines.
 ```suggestion
You can override the identity that was assigned to KEDA during installation, by specifying an `identityId` parameter under the `podIdentity` field. This allows end-users to use different identities to access various resources which is more secure than using a single identity that has access to multiple resources.
```
